helm-chart/todoapp/templates/deployment.yml

```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Release.Name}}-{{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{ .Values.todoapp.rollingUpdate.maxSurge }}
+      maxUnavailable: {{ .Values.todoapp.rollingUpdate.maxUnavailable}}
+  selector:
+    matchLabels:
+      app: {{.Chart.Name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Chart.Name}}
+    spec:
+      serviceAccountName: {{ .Values.todoapp.RBAC.name}}
```

Comment:
Ensure that `{{ .Values.todoapp.RBAC.name }}` is correctly defined in your `values.yaml` file. If not set, this could lead to issues with service account usage.